kernel/fs.c

```diff
@@ -209,15 +209,25 @@ dword_t sys_read(fd_t fd_no, addr_t buf_addr, dword_t size) {
         return _ENOMEM;
     int_t res = 0;
     struct fd *fd = f_get(fd_no);
-    if (fd == NULL || fd->ops->read == NULL) {
+    if (fd == NULL) {
         res = _EBADF;
         goto out;
     }
     if (S_ISDIR(fd->type)) {
         res = _EISDIR;
         goto out;
     }
-    res = fd->ops->read(fd, buf, size);
+    if (fd->ops->pread) {
+        res = fd->ops->pread(fd, buf, size, fd->offset);
```

Comment:
This assigns an ssize_t to an int_t, I'd be worried about truncation

Reply:
Both `pread(2)` and `read(2)` return a `ssize_t`, so I just kept the type that was there before:

https://github.com/tbodt/ish/blob/3ca3bf672a4cf2e0bc1a76e8eca83aef47d3d79b/kernel/fs.c#L220

What should I be using? `int_t`?

Reply:
Fair. Let's leave this then.